pyproject.toml

```diff
@@ -1,6 +1,6 @@
 [build-system]
 requires = ["setuptools>=45", "setuptools_scm[toml]>=6.2", "setuptools-rust"]
```

Comment:
> maturin CLI suggests changing build-backend = "maturin", but IIUC, that makes pytest complain

Ref for the point above: https://github.com/dottxt-ai/outlines-core/actions/runs/13075797112/job/36487540734#step:4:1

Seems that pip is trying to use `maturin` as the build backend, but it isn't available in the isolated build environment used by pip, try adding `maturin` to `requires` too and dropping `setuptools-rust`.

Example from: https://github.com/PyO3/maturin?tab=readme-ov-file#source-distribution:
```
[build-system]
requires = ["maturin>=1.0,<2.0"]
build-backend = "maturin"
```